### ISSUE TITLE
Dockerfile build failed

### ISSUE THREAD

**Issue by BobRam** [dbmail/dbmail#280]
Hi,

docker build fail for me with many errors one by one, the last one is

```
PASS: check_dbmail_common
PASS: check_dbmail_server
PASS: check_dbmail_deliver
FAIL: check_dbmail_imapd
FAIL: check_dbmail_message
PASS: check_dbmail_mailbox
PASS: check_dbmail_mailboxstate
PASS: check_dbmail_auth
PASS: check_dbmail_misc
FAIL: check_dbmail_list
FAIL: check_dbmail_mempool
PASS: check_dbmail_user
PASS: check_dbmail_util
PASS: check_dbmail_dsn
FAIL: check_dbmail_capa
FAIL: check_dbmail_sset
FAIL: check_dbmail_string
PASS: check_dbmail_db
============================================================================
Testsuite summary for dbmail 3.5.0
============================================================================
# TOTAL: 18
# PASS:  11
# SKIP:  0
# XFAIL: 0
# FAIL:  7
# XPASS: 0
# ERROR: 0
============================================================================
See test/test-suite.log
Please report to dbmail@dbmail.org
============================================================================
make[3]: *** [Makefile:3532: test-suite.log] Error 1
make[3]: Leaving directory '/app/test'
make[2]: *** [Makefile:3640: check-TESTS] Error 2
make[2]: Leaving directory '/app/test'
make[1]: *** [Makefile:3831: check-am] Error 2
make[1]: Leaving directory '/app/test'
make: *** [Makefile:2896: check-recursive] Error 1
The command '/bin/sh -c make check' returned a non-zero code: 2
```

Before this i fixed some others:

- created missing folder:  mkdir ./docker/etc
- abuild-keygen -ain   # add "n" - non interactive
- RUN apk add libmhash --repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing/               # libmhash in testing
- RUN apk add libmhash-dev --repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing/        # libmhash in testing
- ARG gmime_APKBUILD=https://gitlab.alpinelinux.org/alpine/aports/-/raw/3.20-stable/community/gmime/APKBUILD  # old link is for gmime-2 not 3
- FROM alpine:3.20 AS build-base # just to fix version...

Could you help me finish my build?

Thank you

**Comment by alan-hicks** [comment]
Thanks for your feedback, it's appreciated, also apologies for taking so long to get to this.

I've updated contrib/Dockerfile.alpine following your comments. I've recently updated the tests so more pass so I have more work to do; all tests pass on FreeBSD where I do most of my development.

I'm not that familiar with Alpine linux but if check_dbmail_mempool fails then I'm not sure how I can help. The following tests failed:

check_dbmail_list
check_dbmail_mempool
check_dbmail_capa
check_dbmail_sset
check_dbmail_string

The alpine Dockerfile has moved to contrib/Dockerfile.alpine so it should be easier to find.

Any feedback welcome.

**Comment by alan-hicks** [comment]
After recent commits all tests pass.

There may be an issue on a busy system with dbmail_check_mailbox timing out, but otherwise all tests pass as of #329.

**Comment by BobRam** [comment]
Hi,
Last try still fail:

69.99 PASS: check_dbmail_common
70.02 PASS: check_dbmail_server
70.58 PASS: check_dbmail_deliver
70.87 FAIL: check_dbmail_imapd
75.37 FAIL: check_dbmail_message
78.38 PASS: check_dbmail_mailbox
78.67 PASS: check_dbmail_mailboxstate
78.71 PASS: check_dbmail_auth
78.84 FAIL: check_dbmail_misc
78.89 PASS: check_dbmail_list
78.93 PASS: check_dbmail_mempool
79.00 PASS: check_dbmail_user
79.06 PASS: check_dbmail_util
79.09 PASS: check_dbmail_util_empty_envelope
79.20 PASS: check_dbmail_dsn
79.23 PASS: check_dbmail_capa
79.29 PASS: check_dbmail_sset
79.34 PASS: check_dbmail_string
79.60 PASS: check_dbmail_db
79.62 ============================================================================
79.62 Testsuite summary for dbmail 3.5.0
79.62 ============================================================================
79.62 # TOTAL: 19
79.62 # PASS:  16
79.62 # SKIP:  0
79.62 # XFAIL: 0
79.62 # FAIL:  3
79.62 # XPASS: 0
79.62 # ERROR: 0
79.62 ============================================================================
79.62 See test/test-suite.log
79.62 Please report to dbmail@dbmail.org
79.62 ============================================================================

**Comment by alan-hicks** [comment]
Can you provide details of the failing tests please: test/test-suite.log

**Comment by BobRam** [comment]
I used the following:

```
cd docker
docker build -t dbmail .
```
so i used the Dockerfile - not Dockerfile.alpine

test-suite.log attached
[test-suite.log](https://github.com/user-attachments/files/18092215/test-suite.log)

PS.
`docker buildx build --tag dbmail:alpine --file docker/Dockerfile.alpine .`
returns:
` => ERROR [base-image  3/13] COPY docker/etc/ /etc/ `
since there is no /etc folder

**Comment by alan-hicks** [comment]
The latest update passed on my machine, hopefully it will on yours too.

#8 116.1 PASS: check_dbmail_common
#8 116.2 PASS: check_dbmail_server
#8 119.5 PASS: check_dbmail_deliver
#8 120.1 PASS: check_dbmail_imapd
#8 171.4 PASS: check_dbmail_message
#8 203.5 PASS: check_dbmail_mailbox
#8 206.6 PASS: check_dbmail_mailboxstate
#8 206.7 PASS: check_dbmail_auth
#8 207.0 PASS: check_dbmail_misc
#8 207.1 PASS: check_dbmail_list
#8 207.1 PASS: check_dbmail_mempool
#8 207.5 PASS: check_dbmail_user
#8 207.6 PASS: check_dbmail_util
#8 207.7 PASS: check_dbmail_util_empty_envelope
#8 208.2 PASS: check_dbmail_dsn
#8 208.2 PASS: check_dbmail_capa
#8 208.3 PASS: check_dbmail_sset
#8 208.4 PASS: check_dbmail_string
#8 210.4 PASS: check_dbmail_db
#8 210.5 ============================================================================
#8 210.5 Testsuite summary for dbmail 3.5.0
#8 210.5 ============================================================================
#8 210.5 # TOTAL: 19
#8 210.5 # PASS:  19
#8 210.5 # SKIP:  0
#8 210.5 # XFAIL: 0
#8 210.5 # FAIL:  0
#8 210.5 # XPASS: 0
#8 210.5 # ERROR: 0
#8 210.5 ============================================================================

**Comment by svalcken** [comment]
i have also try'd to build an docker image.

`docker image build --file Dockerfile-amd64-ubuntu-350 --tag dbmail:latest .`

'34.60 PASS: check_dbmail_common
34.62 PASS: check_dbmail_server
36.88 PASS: check_dbmail_deliver
37.07 FAIL: check_dbmail_imapd
44.02 FAIL: check_dbmail_message
48.54 PASS: check_dbmail_mailbox
48.96 PASS: check_dbmail_mailboxstate
48.99 PASS: check_dbmail_auth
49.08 FAIL: check_dbmail_misc
49.11 PASS: check_dbmail_list
49.13 PASS: check_dbmail_mempool
49.21 PASS: check_dbmail_user
49.25 PASS: check_dbmail_util
49.27 PASS: check_dbmail_util_empty_envelope
49.38 PASS: check_dbmail_dsn
49.40 PASS: check_dbmail_capa
49.45 PASS: check_dbmail_sset
49.48 PASS: check_dbmail_string
49.84 PASS: check_dbmail_db
49.86 ============================================================================
49.86 Testsuite summary for dbmail 3.5.0
49.86 ============================================================================
49.86 # TOTAL: 19
49.86 # PASS:  16
49.86 # SKIP:  0
49.86 # XFAIL: 0
49.86 # FAIL:  3
49.86 # XPASS: 0
49.86 # ERROR: 0
49.86 ============================================================================
49.86 See test/test-suite.log
49.86 Please report to dbmail@dbmail.org
49.86 ============================================================================
49.86 make[3]: *** [Makefile:3583: test-suite.log] Error 1
49.86 make[3]: Leaving directory '/app/dbmail/test'
49.86 make[2]: *** [Makefile:3691: check-TESTS] Error 2
49.86 make[2]: Leaving directory '/app/dbmail/test'
49.86 make[1]: *** [Makefile:3889: check-am] Error 2
49.86 make[1]: Leaving directory '/app/dbmail/test'
49.86 make: *** [Makefile:2926: check-recursive] Error 1'

Environment:
Under Proxmox are an fresh CT-Container with Ubuntu 24.04.2 LTS and Docker version 27.5.1, build 9f9e405

How can i fix it?

**Comment by alan-hicks** [comment]
There is a known issue currently under investigation with gmime 213
You can check your version, it's probably /usr/lib/x86_64-linux-gnu/libgmime*
libgmime-3.0.so.0.213.0 fails a test
libgmime-3.0.so.0.206.1 passes

**Comment by svalcken** [comment]
Right, it is 0.213.0 -> /usr/lib/x86_64-linux-gnu/libgmime-3.0.so -> libgmime-3.0.so.0.213.0

if i understand, the bug is on testing russia and arabica utf8 string compare.
The string will be cutting and then it is not equal. Russia the compare string is 9 chars too short and arabica 15 chars too short.
I think the utf chars with double oder quad byte will not count right.

What can i do to fix this?

Here are the test-logfile:
```
.. contents:: :depth: 2

FAIL: check_dbmail_imapd
========================

Running suite(s): Dbmail Imap
96%: Checks: 54, Failures: 2, Errors: 0
check_dbmail_imapd.c:1340:F:Misc:test_dm_base_subject_misc4:0: Assertion 'res == "привет из россии (hello from russia)"' failed: res == "привет из россии (hello fro", "привет из россии (hello from russia)" == "привет из россии (hello from russia)"
check_dbmail_imapd.c:1355:F:Misc:test_dm_base_subject_utf8:0: Assertion 'res == "مرحبا باللغة العربية (hellow in arabic)"' failed: res == "مرحبا باللغة العربية (h", "مرحبا باللغة العربية (hellow in arabic)" == "مرحبا باللغة العربية (hellow in arabic)"
FAIL check_dbmail_imapd (exit status: 1)

FAIL: check_dbmail_message
==========================

Running suite(s): Dbmail Message
92%: Checks: 25, Failures: 2, Errors: 0
check_dbmail_message.c:629:F:Message:test_dbmail_message_to_string:0: Assertion 'expect == result' failed: expect == "X-Envelope-From: <nobody@pacific.net.sg>
Received: from [127.0.0.1] (port=49353 helo=test11)
        by centos.nowhere.com with smtp (Exim 4.63)
        (envelope-from <nobody@pacific.net.sg>)
        id 1IzWJv-0000Ep-5f
        for wallace@nowhere.com; Tue, 04 Dec 2007 19:52:17 +0800
From: "Wallace" <nobody@pacific.net.sg>
To: wallace <wallace@nowhere.com>
Subject: Test 11
Message-Id: <E1IzWJv-0000Ep-5f@centos.nowhere.com>
Date: Tue, 04 Dec 2007 19:52:16 +0800


This line works, however,
From what I know, this line gets truncated
This line gets truncated
This other line get truncated too
rsio", result == "X-Envelope-From: <nobody@pacific.net.sg>
Received: from [127.0.0.1] (port=49353 helo=test11)
        by centos.nowhere.com with smtp (Exim 4.63)
        (envelope-from <nobody@pacific.net.sg>)
        id 1IzWJv-0000Ep-5f
        for wallace@nowhere.com; Tue, 04 Dec 2007 19:52:17 +0800
From: "Wallace" <nobody@pacific.net.sg>
To: wallace <wallace@nowhere.com>
Subject: Test 11
Message-Id: <E1IzWJv-0000Ep-5f@centos.nowhere.com>
Date: Tue, 04 Dec 2007 19:52:16 +0800


This line works, however,
From what I know, this line gets truncated
This line gets truncated
This other line get truncated too
"
check_dbmail_message.c:1036:F:Message:test_dbmail_message_utf8_headers:0: Assertion 's_dec == t' failed: s_dec == "Пример достаточно дл Subject в многобайтовых символа sortfield формируется без учёта многобайтовых символов, что в свою очередь приводит к ", t == "Пример достаточно дл Subject в многобайтовых символа sortfield формируется без учёта многобайтовых символов, что в свою очередь приводит к"
FAIL check_dbmail_message (exit status: 1)

FAIL: check_dbmail_misc
=======================

Running suite(s): Dbmail Misc
95%: Checks: 20, Failures: 1, Errors: 0
check_dbmail_misc.c:204:F:Misc:test_dbmail_iconv_str_to_db:0: Assertion 'u8 == u82' failed: u8 == "Управленческий учет в современной компании", u82 == "Управленческий учет"
FAIL check_dbmail_misc (exit status: 1)
```

**Comment by alan-hicks** [comment]
Yes, gmime 213 appears to be the issue. They changed their dependencies.
I need to generate a test case as they don't appear to have a representative one. Once I've done that I'll raise an issue.

I've checked DBMail and this error doesn't change the email, just cashing. Annoying but not serious.

The issue is tracked as #364

**Comment by alan-hicks** [comment]
Issue raised upstream [gmime](https://github.com/jstedfast/gmime/issues/176)

**Comment by alan-hicks** [comment]
The gmime issue was with barfed encoding that somehow worked, refactoring it fixed it.
I've refactored the tests as a few of the mime encoded parts weren't valid and created a couple of Dockerfiles [Dockerfile-amd64-ubuntu2404-350](https://github.com/dbmail/dbmail/blob/main/docker/Dockerfile-amd64-ubuntu2404-350) and [Dockerfile-amd64-ubuntu-devel](https://github.com/dbmail/dbmail/blob/main/docker/Dockerfile-amd64-ubuntu-devel) that currently work. You can see how they can be used here [DBMAil/Docker](https://dbmail.org/architecture/docker/)

**Comment by alan-hicks** [comment]
Closing as no feedback and superseded with tests passing on dbmail 3.5.2
